tag/map.go

```diff
@@ -95,13 +100,14 @@ type Mutator interface {
 // Insert returns a mutator that inserts a
 // value associated with k. If k already exists in the tag map,
 // mutator doesn't update the value.
-func Insert(k Key, v string) Mutator {
+// Metadata applies metadata to the tag. It is optional.
```

Comment:
What if there're more than one metadatas and they're conflicting? E.g `Insert(k, v, unlimitedPropagation, noPropagation)`?

Reply:
the last one will be effective.

Reply:
Consider adding a note on the API document, and adding a few tests on this case.

Reply:
added a note in the apis and testcases.